django/http/request.py

```diff
@@ -455,8 +461,19 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
             except UnicodeDecodeError:
                 # ... but some user agents are misbehaving :-(
                 query_string = query_string.decode('iso-8859-1')
-        for key, value in limited_parse_qsl(query_string, **parse_qsl_kwargs):
-            self.appendlist(key, value)
+        try:
+            for key, value in parse_qsl(query_string, **parse_qsl_kwargs):
+                self.appendlist(key, value)
+        except ValueError as e:
+            # Note that ValueError can also be raised if the strict_parsing
+            # argument to parse_qsl() is True. As that is not used by Django,
+            # we assume that the exception was raised by exceeding the value of
+            # max_num_fields instead of fragile checks of exception message
+            # strings.
```

Comment:
I also added this comment based on a previous review from @jdufresne. I feel it is worth mentioning this here.

Reply:
It's OK, I ❤️ comments 😀